OpenImis.ModulesV2/InsureeModule/Repositories/FamilyRepository.cs

```diff
@@ -0,0 +1,273 @@
+﻿using System;
+using System.Data;
+using System.Linq;
+using System.Data.SqlClient;
+using System.Collections.Generic;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using OpenImis.DB.SqlServer;
+using OpenImis.ModulesV2.Helpers;
+using OpenImis.ModulesV2.InsureeModule.Models;
+using OpenImis.ModulesV2.InsureeModule.Models.EnrollFamilyModels;
+using System.Diagnostics;
+using OpenImis.ModulesV2.InsureeModule.Helpers;
+using System.Data.Common;
+using System.IO;
+using Microsoft.AspNetCore.Hosting;
+using System.Xml;
+
+namespace OpenImis.ModulesV2.InsureeModule.Repositories
+{
+    public class FamilyRepository : IFamilyRepository
+    {
+        private IConfiguration _configuration;
+        private readonly IHostingEnvironment _hostingEnvironment;
+
+        public FamilyRepository(IConfiguration configuration, IHostingEnvironment hostingEnvironment)
+        {
+            _configuration = configuration;
+            _hostingEnvironment = hostingEnvironment;
+        }
+
+        public FamilyModel GetByCHFID(string chfid, int userId)
```

Comment:
If userUUID in use in controller then we can change here to userUUID. Less DB access.

Reply:
In this case, you still need to get userId because it is needed in this query:

```
var locationIds = imisContext.TblUsersDistricts
                        .Where(d => d.UserId == userId && d.ValidityTo == null)
                        .Select(x => x.LocationId).ToList();
```

`TblUsersDistricts` contains only `UserId`

**Another solution might be something like this:**

```
var locationIds = (from UD in imisContext.TblUsersDistricts
                   join U in imisContext.TblUsers on UD.UserId equals U.UserId
                   where U.UserUUID == userUUID && UD.ValidityTo == null
                   select UD.LocationId)
                   .ToList();
```

Reply:
Go for the second solution.

Reply:
Done